File: Lists/01_Remove_Kth_Last_Element_From_LL/Cpp/main.cpp
#include <iostream>

#include "remove_kth.hpp"

int main(int artgc, char** argv){
    Node* head = new Node(1);
    Node* runner = head;

    for(int i = 1; i < 11; i++){
        runner->next = new Node(i+1);
        runner = runner->next;
    }

    runner = head;
    std::cout << "->";
    for(; runner != NULL; runner = runner->next){
        std::cout << runner->val << "->";
    }
    std::cout << "/" << std::endl;

    Node* temp = remove_kth_element_suboptimal(head,2);
    Node* temp2 = remove_kth_element_suboptimal(head,2);

    runner = temp;
    std::cout << "->";
    for(; runner != NULL; runner = runner->next){
        std::cout << runner->val << "->";
    }
    std::cout << "/" << std::endl;

    runner = temp2;
    std::cout << "->";
    for(; runner != NULL; runner = runner->next){
        std::cout << runner->val << "->";
    }
    std::cout << "/" << std::endl;

    return 0;

}

File: Lists/01_Remove_Kth_Last_Element_From_LL/Python/main.py
class Node():
    def __init__(self, value = 0, next = None):
        self.value = value
        self.next = next

def remove_node(head, n):
    current = head
    length = 0
    while current != None:
        current = current.next
        length += 1

    current = head
    length = length - n
    for _ in range(length - 1):
        current = current.next

    if current.next:
        current.next = current.next.next
    else:
        current.next = None

    return head
